source/common/stats/thread_local_store.h

```diff
@@ -16,6 +16,95 @@
 namespace Envoy {
 namespace Stats {
 
+/**
+ * A histogram that is stored in TLS and used to record values per thread. This holds two
+ * histograms, one to collect the values and other as backup that is used for merge process. The
+ * swap happens during the merge process.
+ */
+class ThreadLocalHistogram : public virtual Histogram {
```

Comment:
It seems strange that this interface is under `source/` rather than `include/`, especially since `ParentHistogram` is under `include/` and its docs implicitly reference the thread local histograms.

Reply:
It was discussed that we do not want to expose the ThreadLocalHistogram in the stats interface as it is implementation detail. I would adjust the docs of ParentHistogram not to refer this in stats.h definition.

Reply:
That makes sense.  But in that same vein, why do we need to expose `ParentHistogram` in the interface?

Reply:
Ahhh, I see.  Thanks for the explanation.  Sorry to keep pulling at this, as it's really just a nit, but I do have one last question - why do we need the abstract base class?  It seems that all uses (including `TlsHistogramSharedPtr`) explicitly reference the `ThreadLocalHistogramImpl`.  Do you think that we'll ultimately have other TLS histogram implementations?  If so, maybe we should use the base class in those places?

Reply:
I think you can just remove this interface entirely. Is it even needed? I would just fold it all into the implementation below which is specific to this thread_local_store implementation?

Reply:
@mattklein123 +1

Reply:
Yes. We can do that. I have changed. PTAL.